File: frontend/console/src/api/configure.js
import Axios from "@/http";

export function createConfigure(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/configure', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveConfigure(configureId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/configure/${ configureId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateConfigure(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/configure', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteConfigure(configureId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/configure/${ configureId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteConfigureWithFlag(configureId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/configure/${ configureId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllConfigures() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/configure/all').then(res => {
            resolve(res)
        })
    })
}

export function queryConfigure (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/configure/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/configure.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_CONFIGURE = "/config/configure"
const URL_ROUTER_CONFIGURE_BY_ID = "/config/configure/:configure-id"
const URL_ROUTER_CONFIGURE_QUERY = "/config/configure/query"
const URL_ROUTER_CONFIGURE_ALL = "/config/configure/all"

func ConfigureRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_CONFIGURE, config.CreateConfigure)
	r.GET(URL_ROUTER_CONFIGURE_BY_ID, config.RetrieveConfigure)
	r.PUT(URL_ROUTER_CONFIGURE, config.UpdateConfigure)
	r.DELETE(URL_ROUTER_CONFIGURE_BY_ID, config.DeleteConfigure)
	r.PUT(URL_ROUTER_CONFIGURE_BY_ID, config.DeleteConfigureWithFlag)
	r.POST(URL_ROUTER_CONFIGURE_QUERY, config.QueryConfigure)
	r.GET(URL_ROUTER_CONFIGURE_ALL, config.GetAllConfigures)
	return r
}